### PR TITLE
Feat/market and locator authentication

### PR DESCRIPTION
**Pull request by oopcode**
* Marketplace uses secure connections
* All components that communicate with Locator and Marketplace now check their ethereum addrs

### CODE REVIEW

File: cmd/marketplace/main.go [hunk 1]
```diff
@@ -6,14 +6,15 @@ import (
...
+	"github.com/sonm-io/core/accounts"
 	"github.com/sonm-io/core/insonmnia/logging"
 	"github.com/sonm-io/core/insonmnia/marketplace"
 	"go.uber.org/zap"
 	"golang.org/x/net/context"
 )
 
 var (
-	listenAddr  = flag.String("addr", ":9095", "Marketplace service listen address")
+	configPath  = flag.String("config", "marketplace.yaml", "Path to locator config file")
```
Comment (antmat) [inline cmd/marketplace/main.go]: copy pasted "locator"

File: insonmnia/marketplace/config.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	cfg := &MarketplaceConfig{}
+	err := configor.Load(cfg, path)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+type EthConfig struct {
```
Comment (antmat) [inline insonmnia/marketplace/config.go]: How many Eth configs do we have in our repo?

File: cmd/marketplace/main.go [hunk 3]
```diff
@@ -26,11 +27,29 @@ func main() {
...
 
+	ctx := context.Background()
+
+	cfg, err := marketplace.NewConfig(*configPath)
+	if err != nil {
+		log.GetLogger(ctx).Error("failed to load config", zap.Error(err))
+		os.Exit(1)
+	}
+
+	key, err := accounts.LoadKeys(cfg.Eth.Keystore, cfg.Eth.Passphrase)
```
Comment (antmat) [inline cmd/marketplace/main.go]: Consider creating generic EthConfig and put processing inside

File: insonmnia/marketplace/config.go (hunk 2 above)
Comment (oopcode) [reply insonmnia/marketplace/config.go]: okay, will do!

File: cmd/marketplace/main.go (hunk 3 above)
Comment (oopcode) [reply cmd/marketplace/main.go]: done

File: cmd/cli/config/config.go [hunk 4]
```diff
@@ -24,21 +24,20 @@ type Config interface {
...
+	Eth       accounts.EthConfig `yaml:"ethereum"`
+	OutFormat string             `required:"false" default:"" yaml:"output_format"`
 }
 
 func (cc *cliConfig) OutputFormat() string {
 	return cc.OutFormat
 }
 
 func (cc *cliConfig) PassPhrase() string {
-	return cc.Passphrase
```
Comment (antmat) [inline cmd/cli/config/config.go]: I'd better use Eth directly or via method (e.g. Eth().Passphrase()), but ok for now

File: accounts/config.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	"crypto/ecdsa"
+)
+
+type EthConfig struct {
+	Passphrase string `required:"false" default:"" yaml:"pass_phrase"`
+	Keystore   string `required:"false" default:"" yaml:"key_store"`
+}
+
+func (c *EthConfig) LoadKey() (*ecdsa.PrivateKey, error) {
+	key, err := LoadKeys(c.Keystore, c.Passphrase)
```
Comment (antmat) [inline accounts/config.go]: Just a thought - I think we need to make some config validation and postprocessing, where we can load keys for example and make some usefull work (not now though)

**Pull request merged**